### ISSUE TITLE
Training Other Language Pairs for Translation

### ISSUE THREAD

**Issue by reubenharry** [pytorch/fairseq#309]
I've had success training the fairseq translation models that have data preparation scripts (German - English, English - German, English - French) but need to train some others (I'd particularly like French-English, English-Spanish, and Spanish-English). Would you recommend I just modify the existing scripts, and if so, where can I find corpora for other language pairs, e.g. Spanish-English, French-English.

Thanks!

**Comment by edunov** [comment]
Yes, you can change the scripts or write your own, what you want ultimately is parallel data tokenized and BPEed in two text files. And then you simply need to run preprocess.py

The trick is to find datasets, I'd suggest looking at previous year's WMT datasets first, e.g. pretty big and widely used En-Fr dataset is a part of WMT'14 http://www.statmt.org/wmt14/translation-task.html 

Paracrawl has a ton of data, although the quality is not always great: https://paracrawl.eu/releases.html

OpenSubtitles2018 is yet another source of data: http://opus.nlpl.eu/OpenSubtitles2018.php

**Comment by reubenharry** [comment]
Thanks, that's very helpful!
